File: articles/future-directions-in-mathematical-modeling/fortran/future_modeling_priority_summary.f90
program future_modeling_priority_summary
  implicit none
  integer, parameter :: n = 5
  character(len=32) :: keys(n)
  real(8) :: complexity(n), maturity(n), governance(n), uncertainty(n), judgment(n), score(n)
  integer :: i

  keys = (/ 'hybrid_models                 ', 'ai_assistance                 ', 'digital_twins                 ', 'uncertainty_workflows         ', 'participatory_modeling        ' /)
  complexity = (/ 0.88d0, 0.82d0, 0.86d0, 0.90d0, 0.78d0 /)
  maturity = (/ 0.70d0, 0.78d0, 0.75d0, 0.72d0, 0.62d0 /)
  governance = (/ 0.74d0, 0.90d0, 0.88d0, 0.82d0, 0.86d0 /)
  uncertainty = (/ 0.72d0, 0.76d0, 0.70d0, 0.92d0, 0.68d0 /)
  judgment = (/ 0.80d0, 0.92d0, 0.84d0, 0.86d0, 0.94d0 /)

  print '(A)', 'key complexity maturity governance uncertainty human_judgment future_priority_score'
  do i = 1, n
    score(i) = 0.25d0*complexity(i) + 0.20d0*maturity(i) + 0.20d0*governance(i) + 0.20d0*uncertainty(i) + 0.15d0*judgment(i)
    print '(A,1X,F8.4,1X,F8.4,1X,F8.4,1X,F8.4,1X,F8.4,1X,F10.4)', trim(keys(i)), complexity(i), maturity(i), governance(i), uncertainty(i), judgment(i), score(i)
  end do
end program future_modeling_priority_summary

File: articles/future-directions-in-mathematical-modeling/c/future_modeling_priority_summary.c
#include <stdio.h>
#include <stdlib.h>

typedef struct { const char *key; double complexity, maturity, governance, uncertainty, judgment; } Direction;

double priority(Direction d) {
    return 0.25*d.complexity + 0.20*d.maturity + 0.20*d.governance + 0.20*d.uncertainty + 0.15*d.judgment;
}

const char *review_class(Direction d) {
    double s = priority(d);
    if (d.governance >= 0.85 || d.judgment >= 0.90) return "governance_priority";
    if (d.uncertainty >= 0.85) return "uncertainty_priority";
    if (s >= 0.78) return "strategic_priority";
    return "monitor";
}

int main(void) {
    Direction rows[] = {
        {"hybrid_models",0.88,0.70,0.74,0.72,0.80},
        {"ai_assistance",0.82,0.78,0.90,0.76,0.92},
        {"digital_twins",0.86,0.75,0.88,0.70,0.84},
        {"uncertainty_workflows",0.90,0.72,0.82,0.92,0.86},
        {"participatory_modeling",0.78,0.62,0.86,0.68,0.94}
    };
    printf("key,complexity_relevance,technical_maturity,governance_need,uncertainty_pressure,human_judgment_need,future_priority_score,review_class\n");
    for (int i = 0; i < 5; ++i) {
        printf("%s,%.6f,%.6f,%.6f,%.6f,%.6f,%.6f,%s\n", rows[i].key, rows[i].complexity, rows[i].maturity, rows[i].governance, rows[i].uncertainty, rows[i].judgment, priority(rows[i]), review_class(rows[i]));
    }
    fprintf(stderr, "future_modeling_priority_summary complete\n");
    return EXIT_SUCCESS;
}
